src/lib/iac/detect-iac.ts

```diff
@@ -90,14 +91,18 @@ async function getDirectoryFiles(
   root: string,
   options: { maxDepth?: number } = {},
 ) {
-  const iacFiles: IacFileInDirectory[] = [];
   const dirPath = pathLib.resolve(root, '.');
   const supportedExtensions = new Set(Object.keys(projectTypeByFileType));
 
   const directoryPaths = makeDirectoryIterator(dirPath, {
     maxDepth: options.maxDepth,
   });
 
+  const iacFiles: IacFileInDirectory[] = [];
+  const maxConcurrent = 50;
+  const queue = new Queue(maxConcurrent);
+
+  const promises: Array<Promise<IacFileInDirectory>> = [];
```

Comment:
nit: The `type[]` is a more common ideom now for expressing an array.

```suggestion
  const promises: Promise<IacFileInDirectory>[] = [];
```

Reply:
Ah, this was my auto TSLint fix.
Apparently it is giving an: 
`TSLint: Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead.(array-type)` [link](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/array-type.md)

From a bit of googling, I understand that **[ ]** and **Array** are the same, except for the times we are using a `readonly` type. [reference](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#caveats)

I'm happy to put it back as it was since this is the preference, just wondering if I should be adding an ignore rule in TSLint too:
`"rules": {
    "array-type": false
}`

Reply:
Oh my apologies, leave it as is.